File: 02 Even Fibonacci numbers/problem2.js
evenFibs = function(limit) {
  var current = 2;
  var prev = 1;

  var evenFibs = [current];

  while(current < limit) {
    next = current + prev;

    if (next % 2 === 0){evenFibs.push(next);}
    prev = current;
    current = next;
  }

  return evenFibs.reduce(function(a,b){return a+b;});
}

console.log(evenFibs(4000000));

File: 02 Even Fibonacci numbers/problem2.swift
func test(max: Int) -> Int{
  var current = 2
  var prev = 1
  var sum = 2
  
  while current < max {
    var next = current + prev
    
    if next%2==0 {
        sum += next
    }
    prev = current
    current = next
  }
  return sum
}
  println(test(4000000))
